common/phal_L4/spi/spi.c

```diff
@@ -175,6 +247,84 @@ void DMA1_Channel2_IRQHandler()
         DMA1->IFCR |= DMA_IFCR_CGIF2;
     }
 }
+void DMA2_Channel1_IRQHandler()
+{
+    if (DMA2->ISR & DMA_ISR_TEIF2)
+    {
+        DMA2->IFCR |= DMA_IFCR_CTEIF2;
+        if (active_transfer)
+            active_transfer->_error = true;
+    }
+    if (DMA2->ISR & DMA_ISR_TCIF2)
+    {
+        if (active_transfer->nss_sw)
+            PHAL_writeGPIO(active_transfer->nss_gpio_port, active_transfer->nss_gpio_pin, 1);
+
+        // Disable DMA channels
+        PHAL_stopTxfer(active_transfer->rx_dma_cfg);
+        PHAL_stopTxfer(active_transfer->tx_dma_cfg);
+
+        // Revert to mem_inc if trash_can used
+        if (active_transfer->rx_dma_cfg->mem_inc)
+            active_transfer->rx_dma_cfg->channel->CCR |= DMA_CCR_MINC;
+        if (active_transfer->tx_dma_cfg->mem_inc)
+            active_transfer->tx_dma_cfg->channel->CCR |= DMA_CCR_MINC;
+
+        // Disable SPI peripheral and DMA requests
+        SPI3->CR1 &= ~SPI_CR1_SPE;
+        SPI3->CR2 &= ~(SPI_CR2_TXDMAEN | SPI_CR2_RXDMAEN);
+
+        active_transfer->_busy  = false;
+        active_transfer->_error = false;
+        active_transfer = NULL;
+    }
+    if (DMA2->ISR & DMA_ISR_GIF2)
+    {
+        DMA2->IFCR |= DMA_IFCR_CGIF2;
+    }
+}
+/**
+ * @brief SPI1 Rx DMA Transfer Complete interrupt
+ *
+ */
+#ifdef SPI2
+    void DMA1_Channel4_IRQHandler()
+    {
+        if (DMA1->ISR & DMA_ISR_TEIF4)
+        {
+            DMA1->IFCR |= DMA_IFCR_CTEIF4;
+            if (active_transfer)
+                active_transfer->_error = true;
+        }
+        if (DMA1->ISR & DMA_ISR_TCIF4)
+        {
+            if (active_transfer->nss_sw)
```

Comment:
There is a lot of the same code being put into these IRQs. Maybe consider putting the similar stuff in one function and passing the stuff that differs through arguments. Then if someone wants to make an edit, they only have to do it in one location.,